File: 2389_longest_subsequence_with_limited_sum.cpp
class Solution {
public:
    vector<int> answerQueries(vector<int>& n, vector<int>& q) {
        vector<int> v;
        sort(n.begin(),n.end());
        for(int i =0;i<q.size();i++){
            int x = q[i];
            int sum =0;
            int count =0;
            for(int j = 0;j<n.size();j++){
                sum+=n[j];
                if(sum<=x){
                    count++;
                }
            }
            v.push_back(count);
        }
        return v;
    }
};

File: 2389-longest-subsequence-with-limited-sum/2389-longest-subsequence-with-limited-sum.java
class Solution {
    public int[] answerQueries(int[] nums, int[] queries) {
        Arrays.sort(nums);
        int n=queries.length;
        int[]ans=new int[n];
        for(int i=0;i<n;i++){
            int sum=0;
            int j=0;
            for(;j<nums.length;j++){
                if(sum+nums[j]>queries[i])break;
                sum+=nums[j];
            }
            ans[i]=j;
        }
        return ans;
    }
}